src/backend/plugins/video/warren_video/indicators/daily_video_views.py

```diff
@@ -0,0 +1,99 @@
+"""Daily Video Views class.
+
+Defines the daily video views indicator. It is, for a given video, and
+a date range, the total number of views, and the number of views per day.
+"""
+from typing import List
+
+import pandas as pd
+from ralph.backends.http.lrs import BaseHTTP, LRSQuery
+from ralph.models.xapi.concepts.constants.video import RESULT_EXTENSION_TIME
+from ralph.models.xapi.concepts.verbs.video import PlayedVerb
+from warren_video.conf import settings as video_plugin_settings
+from warren_video.indicators.models import VideoViews
+
+from warren.base_indicator import BaseIndicator
+from warren.filters import DatetimeRange
+
+
+class DailyVideoViews(BaseIndicator):
+    """An indicator for the daily video views metric.
+
+    Args:
+        client: The LRS backend from which the query is to be issued
+        video_uuid: The UUID of the video on which to compute the metric
+        date_range: The date range on which to compute the indicator. It has
+            2 fields, `since` and `until` which are dates or timestamps that must be
+            in ISO format (YYYY-MM-DD or YYYY-MM-DDTHH:mm:ss.SSSZ")
+    """
+
+    client: BaseHTTP
+    video_uuid: str
+    date_range: DatetimeRange
+
+    def __init__(self, client: BaseHTTP, video_uuid: str, date_range: DatetimeRange):
+        """Instantiate the indicator with its parameters."""
+        self.client = client
+        self.video_uuid = video_uuid
+        self.date_range = date_range
+
+    def get_lrs_query(self) -> LRSQuery:
+        """Returns the LRS query for fetching required statements."""
+        query = {
+            "verb": PlayedVerb().id,
+            "activity": self.video_uuid,
+            "since": self.date_range.since.isoformat(),
+            "until": self.date_range.until.isoformat(),
+        }
+        return LRSQuery(query=query)
+
+    def fetch_statements(self) -> List:
+        """Executes the LRS query to obtain statements required for this indicator."""
+        filter_query = self.get_lrs_query()
+        statements = list(
+            self.client.read(target=self.client.statements_endpoint, query=filter_query)
+        )
+        return statements
+
+    def compute(self) -> VideoViews:
+        """Computes the current indicator.
+
+        Gets the statements from the LRS, filters and aggregate to return the number of
+        video views per day.
+        """
+        response = VideoViews()
```

Comment:
I you keep in mind that this indicator will be used outside from an HTTP context (with the request/response semantic), I would rename this variable as `indicator`. WDYT?

```suggestion
        indicator = VideoViews()
```